@here/harp-datasource-protocol/lib/ExprPool.ts

```diff
@@ -33,17 +42,24 @@ export class ExprPool implements ExprVisitor<Expr, void> {
     private readonly m_varExprs = new Map<string, VarExpr>();
     private readonly m_hasAttributeExprs = new Map<string, HasAttributeExpr>();
     private readonly m_inExprs = new Map<Expr, ContainsExpr[]>();
+    private readonly m_matchExprs: MatchExpr[] = [];
+    private readonly m_caseExprs: CaseExpr[] = [];
     private readonly m_callExprs = new Map<string, CallExpr[]>();
 
+    private nextKey: number = 1;
     /**
      * Add `expr` to this [[ExprPool]] and return a unique [[Expr]]
      * object that is structurally equivalent to `expr`.
      *
-     * @param expr The [[Expr]] to add to this [[ExprPool]].
+     * @param expr The [[PooledExpr]] to add to this [[ExprPool]].
```

Comment:
Minor typo: `@param` is an `Expr` and `@return` is a `PooledExpr`

Reply:
Fixed, thanks.